File: Lighthouse.Frontend/src/models/Preview/PreviewFeature.ts
export interface IPreviewFeature {
    name: string;
    key: string;
    id: number;
    description: string;
    enabled: boolean;
}

export class PreviewFeature implements IPreviewFeature {
    name: string;
    key: string;
    id: number;
    description: string;
    enabled: boolean;

    constructor(id: number, key: string, name: string, description: string, enabled: boolean) {
        this.name = name;
        this.key = key;
        this.id = id;
        this.description = description;
        this.enabled = enabled;
    }
}

File: Lighthouse.Backend/Lighthouse.Backend/Models/Preview/PreviewFeature.cs
using Lighthouse.Backend.Services.Interfaces;

namespace Lighthouse.Backend.Models.Preview
{
    public class PreviewFeature : IEntity
    {
        public required int Id { get; set; }

        public string Key { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public required bool Enabled { get; set; }
    }
}
